### DISCUSSION TITLE
Project community metrics

### DISCUSSION THREAD

**Discussion by dalers** [timschofield/webERP#308] [Development Discussion & Specification]
@timschofield you said you liked data.....  

Every time I log into the demo it seems to be in a different language. It might be interesting to create a metric for number of times the language is changed in a day/hour/month and plot a heatmap for number of times vs language (and maybe how many times the language is changed to a particular language, and how many logins there that didn't change the language from what it was last set to.... lots of fun numbers :-)  ;-)

Do the GitHub metrics give any insights? https://github.com/timschofield/webERP/pulse

Btw, you have ignited a fire! :-)

Last 30 days:

![image](https://github.com/user-attachments/assets/ea4540d1-1328-4fa1-843f-7bd9f29c5d91)

Last year:

![image](https://github.com/user-attachments/assets/8aa6ac9e-cf09-4c45-bf91-cce1353040c7)

**Comment by dalers** [comment]
Similar metric for themes on the demo site as language.

The more logins that left the theme as they found it, or changed it from one particular theme to another particular theme, might give an insight into the order of popularity of themes. Also cross referenging with language might show interesting characteristics or patterns.

**Comment by timschofield** [comment]
It did occur to me to put the IP address in the audit trail which
would produce interesting data. Not sure how that

On Fri, 13 Dec 2024 at 17:47, Dale Scott ***@***.***> wrote:
>
> Similar metric for themes on the demo site as language.
>
> The more logins that left the theme along, or changed it from a particular theme, or changed it to a partular theme might give an insight into the order of popularity for a theme. Cross referenging with language might also show interesting characteristics or patterns.
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.Message ID: ***@***.***>



-- 
www.weberp.org
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Comment by pakricard** [comment]
I coded a Script Audit that tracks the usage of all scripts, to track CPU usage time and script optimization, but could be used for the purpose @dalers mentioned:

<img width="788" alt="image" src="https://github.com/user-attachments/assets/c3f3d781-b5bf-4214-b933-aba14ea58363" />
<img width="608" alt="image" src="https://github.com/user-attachments/assets/939dfce5-33af-4789-b037-3799b63e366d" />
<img width="572" alt="image" src="https://github.com/user-attachments/assets/fcc53bd0-7401-4204-88c5-a9c8f4ada48c" />

Today it marked the milestone of being 400 commits behind timschofield/webERP master code ;-) . getting closer!

<img width="230" alt="image" src="https://github.com/user-attachments/assets/f5a091f3-feef-473e-a213-bcbbee2491b5" />

If you think my Script Audit could do the trick, I will commit it as soon as possible.

**Reply by dalers** [reply]
Hi @pakricard 

> I coded a Script Audit that tracks the usage of all scripts, to track CPU usage time and script optimization,

Thie script looks REALLY useful to understand user interaction (which is why I expect you coded it!).

- How does it capture the data? Do you have to add metric capture to the other scripts?
- Do you have a menu item in your system for the script? If so, where?

I would definately appreciate you committing the script, perghaps push it to a feature branch in your weberp your fork for us to pull, test and discuss, before sending a pull request for merging into the master branch in the project repo.
